kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java

```diff
@@ -0,0 +1,452 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.jdbc.hive;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.nio.charset.StandardCharsets;
+import java.sql.*;
+import java.util.List;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.hive.service.rpc.thrift.TTableSchema;
+import org.apache.hive.service.rpc.thrift.TTypeId;
+import org.apache.kyuubi.jdbc.hive.adapter.SQLResultSet;
+import org.apache.kyuubi.jdbc.hive.arrow.ArrowColumnarBatchRow;
+import org.apache.kyuubi.jdbc.hive.arrow.ArrowUtils;
+
+/** Data independent base class which implements the common part of all Kyuubi result sets. */
+public abstract class KyuubiArrowBasedResultSet implements SQLResultSet {
+
+  protected Statement statement = null;
+  protected SQLWarning warningChain = null;
+  protected boolean wasNull = false;
+  protected List<String> columnNames;
+  protected List<String> normalizedColumnNames;
+  protected List<TTypeId> columnTypes;
+  protected List<JdbcColumnAttributes> columnAttributes;
+  private TTableSchema schema;
+
+  // arrow related
+  protected Schema arrowSchema;
+  protected VectorSchemaRoot root;
+  protected ArrowColumnarBatchRow row;
+
+  protected BufferAllocator allocator;
+
+  protected void initArrowSchemaAndAllocator() {
+    if (arrowSchema == null) {
+      throw new IllegalStateException("arrow schema is null");
+    }
+    allocator = ArrowUtils.rootAllocator.newChildAllocator("ArrowResultSet", 0, Long.MAX_VALUE);
+    root = VectorSchemaRoot.create(arrowSchema, allocator);
+  }
+
+  @Override
+  public int findColumn(String columnName) throws SQLException {
+    int columnIndex = 0;
+    boolean findColumn = false;
```

Comment:
```suggestion
    boolean found= false;
```

Reply:
since the method is forked from `KyuubiBaseResultSet::findColumn`, I prefer to leave it as the current name.

Reply:
so we have the opportunity to reuse more code?

Reply:
yep